### PR TITLE
Add Makefile to support Homebrew

### PR DESCRIPTION
**Pull request by neakor**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+BINARY_FOLDER_PREFIX?=/usr/local
```
Comment (rudro) [inline Makefile]: Why the `?` ?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+BINARY_FOLDER=$(BINARY_FOLDER_PREFIX)/bin/
+SWIFT_BUILD_FLAGS=-c release -Xswiftc -static-stdlib
+
+.PHONY: clean build install uninstall
+
+clean:
+	cd Generator && swift package clean
```
Comment (rudro) [inline Makefile]: nit: `Generator` is in 3 places.  Maybe a `GENERATOR_SUBDIR` ?

File: Makefile (hunk 1 above)
Comment (neakor) [reply Makefile]: It's the conditional variable assignment thing. We'll pass the `#{prefix}` from the Homebrew script to our Makefile. This is just saying if we don't pass it in, we default to `/usr/local`.

File: Makefile (hunk 2 above)
Comment (neakor) [reply Makefile]: Good point. I'll update that.

**Pull request merged**